components/script/dom/htmlimageelement.rs

```diff
@@ -1366,13 +1400,35 @@ impl MicrotaskRunnable for ImageElementMicrotask {
             } => {
                 elem.react_to_environment_changes_sync_steps(*generation);
             },
+            ImageElementMicrotask::DecodeTask {
```

Comment:
for this one, let's follow the pattern of the other tasks and put the handler into a method on `elem`.